### PR TITLE
add EnvironmentPlugin to webpack

### PR DESCRIPTION
**Pull request by mohsen1**
https://webpack.js.org/plugins/environment-plugin/

### CODE REVIEW

**Comment by sandersn** [comment]
@tkqubo, @mattlewis92, @bumbleblym, @bcherny, @tommytroylin can you take a look?

**Comment by mohsen1** [comment]
ping

**Comment by sandersn** [comment]
I'll wait a couple more days and merge. In the meantime, @mohsen1 can you add a test for the new plugin?

**Comment by bcherny** [comment]
LGTM

**Review by bcherny — APPROVED** [review]

File: types/webpack/webpack-tests.ts [hunk 1]
```diff
@@ -421,6 +421,8 @@ plugin = new webpack.WatchIgnorePlugin(paths);
 plugin = new webpack.LoaderOptionsPlugin({
     debug: true
 });
+plugin = new webpack.EnvironmentPlugin(['a', 'b']);
+plugin = new webpack.EnvironmentPlugin([{a: true, b: 'c'});
```
Comment (sandersn) [inline types/webpack/webpack-tests.ts]: Looks like a missing `]` here

**Comment by sandersn** [comment]
Travis failed because of a typo. Looks good to merge after that's fixed.

File: types/webpack/webpack-tests.ts (hunk 1 above)
Comment (mohsen1) [inline types/webpack/webpack-tests.ts]: Actually extra [ 😂

**Pull request merged**